src/validation.h

```diff
@@ -1038,6 +1038,9 @@ class ChainstateManager
     /** Produce the necessary coinbase commitment for a block (modifies the hash, don't call for mined blocks). */
     std::vector<unsigned char> GenerateCoinbaseCommitment(CBlock& block, const CBlockIndex* pindexPrev) const;
 
+    //! Returns true if any chainstate in use is in initial block download.
+    bool IsAnyChainInIBD() EXCLUSIVE_LOCKS_REQUIRED(::cs_main);
```

Comment:
Not entirely clear to me that this ("Until we have completed IBD on all chainstates, don't answer getheaders requests from our peers or advertise ourselves.") is a good idea? If you've caught up on the snapshot chain, is the fact that you got there via assumeutxo any worse than getting there via assumevalid? No objection to keeping it purely for "first, do no harm" reasons, but if we're already confident there's no harm possible, better to have more nodes sharing headers etc, than less.

Reply:
Good observation. I definitely agree that we should still be answering getheaders requests, which I think we are per this code: https://github.com/jamesob/bitcoin/blob/39ddd522c37f760c8054d2fa58e026c69a50ce8b/src/net_processing.cpp#L3292-L3321. In terms of advertising ourselves while background-validating, I'll remain conservative in this changeset and defer to others... should be a pretty easy follow-up if deemed safe, and it'd allow someone besides me to make assumeutxo-related changes :).